### DISCUSSION TITLE
Is there a way to keep a global state between all the tests (in different files)

### DISCUSSION THREAD

**Discussion by real-felix** [avajs/ava#3011] [Q&A] [answered]
My tests are run against a local service, and they need each a unique ID (from 0 to 255). I use something like that:

```ts
const getNextIndex = (function (n) {
  return () => n++;
})(0);
```

Unfortunately, it looks like each file is run in its own process, so several test are run with the same ID and conflict each other.

Is there an easy way to go around this?

**Comment by novemberborn** [comment, marked as answer]
Yes! Through the [shared workers](https://github.com/avajs/ava/blob/main/docs/recipes/shared-workers.md) feature.

See 	https://github.com/avajs/get-port for an example, that relies on https://github.com/avajs/get-port which lets you reserve a range of integers which no other test worker can then use until the first worker completes.
